docs/csharp/how-to/modify-string-contents.md

```diff
@@ -0,0 +1,74 @@
+
+[!code-csharp-interactive[remove text](../../../samples/snippets/csharp/how-to/strings/ModifyStrings.cs#4)]  
+
+## Replace patterns
+
+You can use [regular expressions](../../standard/base-types/regular-expressions.md) to replace patterns with new patterns. The following example makes use of the <xref:System.Text.RegularExpressions.Regex?displayProperty=nameWithType> class to find a pattern in a source string and replace it with proper capitalization. The <xref:System.Text.RegularExpressions.Regex.Replace%2A?displayProperty=nameWithType> method takes a function that provides the logic of the replacement as one of its arguments. That function uses the <xref:System.Text.StringBuilder?displayProperty=nameWithType> class to build the replacement string with proper capitalization.
+
+The search pattern, "the\s" searches for the word "the" followed by a whitespace character. That part of the pattern ensures that it doesn't match "there" in the source string. Regular expressions are most useful for searching and replacing text that follows a pattern, rather than known text. See [How to: search strings](search-strings.md) for more details.
+
+[!code-csharp-interactive[replace creates a new string](../../../samples/snippets/csharp/how-to/strings/ModifyStrings.cs#5)]  
+
+The <xref:System.Text.StringBuilder.ToString%2A?displayProperty=nameWithType> method returns an immutable string with the contents in the <xref:System.Text.StringBuilder> object.
+
+## Modifying individual characters
+
+You can produce a character array from a string, modify the contents of the array, and then create a new string from the modified contents of the array. 
+
+The following example shows how to replace a set of characters in a string. First, it uses the <xref:System.String.ToCharArray?displayProperty=nameWithName> method to create an array of characters. Then, it uses the <xref:System.String.IndexOf%2A> method to find the starting index of the word "fox." Then, the next three characters are replaced with a different word. Finally, a new string is constructed from the updated character array.
+
+[!code-csharp-interactive[replace creates a new string](../../../samples/snippets/csharp/how-to/strings/ModifyStrings.cs#6)]  
+
+## Unsafe modifications to string
+
+Using **unsafe** code, you can modify a string "in place" after it has been created. Unsafe code bypasses many of the features of .NET designed to minimize certain types of bugs in code. 
+The following example is provided for those rare situations where you want to modify a string in-place using unsafe code. The example shows how to access the individual characters "in-place" by using the fixed keyword. It also demonstrates one possible side effect of unsafe operations on strings that results from the way that the C# compiler stores (interns) strings internally. In general, you shouldn't use this technique unless it is absolutely necessary.
```

Comment:
I'd be somewhat more precise here. Fixed doesn't permit access, it just prevents GC from moving the object you might want to access.